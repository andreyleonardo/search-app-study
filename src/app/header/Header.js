import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import LanguageSelector from './language-selector/LanguageSelector';

import './Header.scss';

const Header = ({ selectedLanguage, onChangeLanguage }) => (
  <header className="header">
    <div className="header__container row middle-xs">
      <div className="header__title col-xs">
        <FormattedMessage id="header.title" />
      </div>
      <nav className="header__nav_bar col-xs">
        <LanguageSelector
          selectedLanguage={selectedLanguage}
          onClick={onChangeLanguage}
        />
      </nav>
    </div>
  </header>
);

Header.propTypes = {
  selectedLanguage: PropTypes.string.isRequired,
  onChangeLanguage: PropTypes.func.isRequired
};

export default Header;
