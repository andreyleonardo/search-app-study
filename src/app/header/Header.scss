@import "src/stylesheets/css/_screen-sizes";

.header {
  overflow: hidden;
  background-color: white;
  position: fixed;
  height: 90px;
  top: 0;
  width: 100%;

  @media screen and (max-width: $mobile_max_width) {
    height: 85px;
  }

  &__container {
    margin: 0;
    padding: 10px;
  }

  &__title {
    font-size: 1.5em;
  }
}
