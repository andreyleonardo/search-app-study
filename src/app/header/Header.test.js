import React from 'react';
import Header from './Header';

describe('Header', () => {
  it('renders without crash', () => {
    const header = shallow(<Header selectedLanguage="pt" onChangeLanguage={jest.fn()} />);

    expect(header).toMatchSnapshot();
  });
});
