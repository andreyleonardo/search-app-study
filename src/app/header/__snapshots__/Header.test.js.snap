// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without crash 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    onChangeLanguage={[MockFunction]}
    selectedLanguage="pt"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="header__container row middle-xs"
      >
        <div
          className="header__title col-xs"
        >
          <FormattedMessage
            id="header.title"
            values={Object {}}
          />
        </div>
        <nav
          className="header__nav_bar col-xs"
        >
          <LanguageSelector
            onClick={[MockFunction]}
            selectedLanguage="pt"
          />
        </nav>
      </div>,
      "className": "header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header__title col-xs"
          >
            <FormattedMessage
              id="header.title"
              values={Object {}}
            />
          </div>,
          <nav
            className="header__nav_bar col-xs"
          >
            <LanguageSelector
              onClick={[MockFunction]}
              selectedLanguage="pt"
            />
          </nav>,
        ],
        "className": "header__container row middle-xs",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FormattedMessage
              id="header.title"
              values={Object {}}
            />,
            "className": "header__title col-xs",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "header.title",
              "values": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <LanguageSelector
              onClick={[MockFunction]}
              selectedLanguage="pt"
            />,
            "className": "header__nav_bar col-xs",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onClick": [MockFunction],
              "selectedLanguage": "pt",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "nav",
        },
      ],
      "type": "div",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="header__container row middle-xs"
        >
          <div
            className="header__title col-xs"
          >
            <FormattedMessage
              id="header.title"
              values={Object {}}
            />
          </div>
          <nav
            className="header__nav_bar col-xs"
          >
            <LanguageSelector
              onClick={[MockFunction]}
              selectedLanguage="pt"
            />
          </nav>
        </div>,
        "className": "header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="header__title col-xs"
            >
              <FormattedMessage
                id="header.title"
                values={Object {}}
              />
            </div>,
            <nav
              className="header__nav_bar col-xs"
            >
              <LanguageSelector
                onClick={[MockFunction]}
                selectedLanguage="pt"
              />
            </nav>,
          ],
          "className": "header__container row middle-xs",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FormattedMessage
                id="header.title"
                values={Object {}}
              />,
              "className": "header__title col-xs",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "id": "header.title",
                "values": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <LanguageSelector
                onClick={[MockFunction]}
                selectedLanguage="pt"
              />,
              "className": "header__nav_bar col-xs",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onClick": [MockFunction],
                "selectedLanguage": "pt",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "nav",
          },
        ],
        "type": "div",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
